File: aegis-lc4j/frontend/src/lib/tenantNames.ts

export function displayName(tenantId: string | null | undefined): string {
  if (!tenantId) return "your bank";
  return tenantId
    .split("-")
    .filter(Boolean)
    .map((p) => p.charAt(0).toUpperCase() + p.slice(1))
    .join(" ");
}

File: aegis-merged/src/main/java/com/aegis/merged/assistant/TenantNames.java
package com.aegis.merged.assistant;

final class TenantNames {

    private TenantNames() {
    }

    static String displayName(String tenantId) {
        if (tenantId == null || tenantId.isBlank()) return "your bank";
        String[] parts = tenantId.split("-");
        StringBuilder sb = new StringBuilder();
        for (String p : parts) {
            if (p.isEmpty()) continue;
            if (!sb.isEmpty()) sb.append(' ');
            sb.append(Character.toUpperCase(p.charAt(0))).append(p.substring(1));
        }
        return sb.toString();
    }
}
